src/main.ts

```diff
@@ -68,6 +68,26 @@ export async function run(context: Context): Promise<void> {
     const updaterImage =
       params.updaterImage || updaterImageName(details['package-manager'])
 
+    // The sendMetrics function is used to send metrics to the API client.
+    // It uses the package manager as a tag to identify the metric.
+    const sendMetricsWithPackageManager: MetricReporter = async (
```

Comment:
I have added this wrapper to inject the `sendMetricsWithPackageManager` with `package_name` and then pass it to the `ImageService.pull`. This was purely done to avoid passing additional parameters to the ImageService.Pull`.